core/src/text_serializer.cc

```diff
@@ -16,9 +17,14 @@ void WriteValue(std::ostream& out, double value) {
   } else if (std::isinf(value)) {
     out << (value < 0 ? "-Inf" : "+Inf");
   } else {
-    auto saved_flags = out.setf(std::ios::fixed, std::ios::floatfield);
+    std::ios oldState{nullptr};
+    oldState.copyfmt(out);
+
+    out.setf(std::ios::fixed, std::ios::floatfield);
+    out << std::setprecision(17);
```

Comment:
Maybe you could use `std::numeric_limits<double>::max_digits10` instead of the magic number:

https://stackoverflow.com/questions/554063/how-do-i-print-a-double-value-with-full-precision-using-cout